fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -119,6 +119,35 @@ ACTOR Future<Void> forwardError(Future<int> input) {
 	throw Error::fromCode(errorCode);
 }
 
+int getWriteStallState(const rocksdb::WriteStallCondition& condition) {
+	if (condition == rocksdb::WriteStallCondition::kDelayed)
+		return 0;
+	if (condition == rocksdb::WriteStallCondition::kStopped)
+		return 1;
+	if (condition == rocksdb::WriteStallCondition::kNormal)
+		return 2;
+
+	// unexpected.
+	return 3;
+}
+
+class RocksDBEventListener : public rocksdb::EventListener {
+public:
+	RocksDBEventListener(UID id) : logId(id) {}
+	void OnStallConditionsChanged(const rocksdb::WriteStallInfo& info) {
```

Comment:
This callback won't be on the main thread, do we know the behavior of `TraceEvent` in this case?

Reply:
TraceEvent is working with RocksDB thread. Histogram and counters will have issues, I think.